### PR TITLE
Impr dockerfile docker way

### PR DESCRIPTION
**Pull request by linuxbandit**
The source files are copied on build time and not mounted.
This is the way the Dockerfiles should be written. Unfortunately
due to scopes, the Dockerfile cannot be inside docker, because
the COPY operation does not go above (..) the Dockerfile position.
Alternative to this could be to 'RUN git clone [....]' but this
approach is too rigid: in the way proposed, the files can still
be modified and the image rebuilt.

### CODE REVIEW

**Comment by linuxbandit** [comment]
Also to be found: what is (if any) the analogous of node_modules

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -3,18 +3,14 @@ version: '3.2'
...
-            dockerfile: ./Dockerfile.dev
-        volumes:
-            - ./$PATH_OMS_CORE_ELIXIR/../:/usr/src/myapp
+            context: ./$PATH_OMS_CORE_ELIXIR/../
         depends_on:
             - postgres-oms-core-elixir
-        expose:
-            - "4000"
+        ports:
+            - 4000:4000
```
Comment (blacksph3re) [inline docker/docker-compose.yml]: I think this is super security risky, I would not expose that directly. I'd rather have it behind the traefik proxy.

**Comment by blacksph3re** [comment]
Cool idea, you can actually use the Dockerfile of 
https://github.com/AEGEE/oms-mailer/blob/master/docker/oms-mailer/Dockerfile
This way only the relevant part of the code will be mounted and also dependency fetching and compilation will happen during build.

File: docker/docker-compose.yml (hunk 1 above)
Comment (linuxbandit) [reply docker/docker-compose.yml]: It was just to have vagrant alert when ready by means of curl (see [here](https://github.com/AEGEE/oms-docker/blob/acdd011b0a1a2b251f92297bbd26198e83cbe4fc/vagrant-post-script/orchestrate_docker.sh))

Comment (blacksph3re) [reply docker/docker-compose.yml]: you can still query it through traefik

Comment (blacksph3re) [reply docker/docker-compose.yml]: also you will not get a html file from core, it's a json api. If you want I can also implement a status endpoint which checks for DB connectivity

Comment (serge1peshcoff) [reply docker/docker-compose.yml]: I support that actually, we can query it through /services/oms-core-elixir/api

Comment (linuxbandit) [reply docker/docker-compose.yml]: Phoenix replies with an HTML saying route not found

I agree with the status endpoint, it can be used by healthchecks

Comment (linuxbandit) [reply docker/docker-compose.yml]: also can we make any api be just `/api`? the url is unnecessarily long

Comment (linuxbandit) [reply docker/docker-compose.yml]: if there's no html, what's [this](https://github.com/AEGEE/oms-core-elixir/tree/master/frontend)?

Comment (serge1peshcoff) [reply docker/docker-compose.yml]: seems like parts of the old frontend (AngularJS one)

**Comment by linuxbandit** [comment]
Check it out

**Comment by linuxbandit** [comment]
superseded

**Pull request closed without merging**